### PR TITLE
Update connection-pooling for Digital Ocean

### PR DESCRIPTION
**Pull request by rowin1125**
Related to Prisma [issue](https://github.com/prisma/prisma/issues/4752)

## Problem
Currently its required to specify some additional connection parameters to be able to deploy via `Vercel`. This information could be really helpful since this is not yet documented for Prisma and Digital Ocean combined with Redwood.js

### CODE REVIEW

**Comment by dac09** [comment]
Hello @rowin1125 - thanks for this - connection pooling can certainly be a mystery. 

I wonder if you could expand a little on the two URLs that digital ocean gives you, marking the parts where it's not clear!

File: docs/docs/connection-pooling.md [hunk 1]
```diff
@@ -65,6 +65,22 @@ Heroku does not officially support MySQL.
 ## Digital Ocean
 For Postgres, see [How to Manage Connection Pools](https://www.digitalocean.com/docs/databases/postgresql/how-to/manage-connection-pools)
 
+In order to run migrations on platforms like `Vercel` and `Netlify` you are required to append additional connection parameters to your database url. Prisma needs to be informed to use pgbouncer (which is part of the connection pool of Ditital Ocean), if not provided the following error will be show:
```
Comment (dac09) [inline docs/docs/connection-pooling.md:68]:
```suggestion
In order to run migrations through a connection pool you are required to append additional connection parameters to your database url. Prisma needs to be informed to use pgbouncer (which is part of the connection pool of Digital Ocean), if not provided the following error will be show:
```

File: docs/docs/connection-pooling.md [hunk 2]
```diff
@@ -65,6 +65,22 @@ Heroku does not officially support MySQL.
...
+Error: Migration engine error:
+db error: ERROR: prepared statement "s0" already exists
+```
+
+To resolve this use the following structure of you database url: 
+```
+<YOUR_CONNECTION_POOL_URL>:25061/defaultdb?connection_limit=3&sslmode=require&pgbouncer=true&connect_timeout=10&pool_timeout=30
+```
+Notes:
+* Make sure to use the correct host and  *NOT* `25060`
```
[the remark this answers is not in the document]
Comment (rowin1125) [reply docs/docs/connection-pooling.md]: Thanks for the feedback. I adjusted to text a little bit. Hopefully its more clear now :)

File: docs/docs/connection-pooling.md [hunk 3]
```diff
@@ -65,6 +65,22 @@ Heroku does not officially support MySQL.
 ## Digital Ocean
 For Postgres, see [How to Manage Connection Pools](https://www.digitalocean.com/docs/databases/postgresql/how-to/manage-connection-pools)
 
+In order to run migrations through a connection pool you are required to append additional connection parameters to your database url. Prisma needs to be informed to use pgbouncer (which is part of the connection pool of Digital Ocean), if not provided you may receive the following error:
```
Comment (jtoar) [inline docs/docs/connection-pooling.md:68]:
```suggestion
To run migrations through a connection pool, you're required to append connection parameters to your `DATABASE_URL`. Prisma needs to know to use pgbouncer (which is part of Digital Ocean's connection pool). If omitted, you may receive the following error:
```

File: docs/docs/connection-pooling.md [hunk 4]
```diff
@@ -65,6 +65,22 @@ Heroku does not officially support MySQL.
...
+
+```
+Error: Migration engine error:
+db error: ERROR: prepared statement "s0" already exists
+```
+
+To resolve this use the following structure of you database url:
```
Comment (jtoar) [inline docs/docs/connection-pooling.md:75]:
```suggestion
To resolve this, use the following structure in your `DATABASE_URL`:

```

File: docs/docs/connection-pooling.md [hunk 5]
```diff
@@ -65,6 +65,22 @@ Heroku does not officially support MySQL.
...
+```
+Error: Migration engine error:
+db error: ERROR: prepared statement "s0" already exists
+```
+
+To resolve this use the following structure of you database url: 
+```
+<YOUR_CONNECTION_POOL_URL>:25061/defaultdb?connection_limit=3&sslmode=require&pgbouncer=true&connect_timeout=10&pool_timeout=30
+```
+Notes:
```
Comment (jtoar) [inline docs/docs/connection-pooling.md:79]:
```suggestion
Here's a couple more things to be aware of:
```

File: docs/docs/connection-pooling.md [hunk 6]
```diff
@@ -65,6 +65,22 @@ Heroku does not officially support MySQL.
...
+- When using a connection pool of Digital Ocean you will have multiple port available.Typically the direct connection (without connection pooling) is on port 25060 and the connection through pgbouncer is served through port 25061. Make sure you connect to your connection pool on port **25061**
```
Comment (jtoar) [inline docs/docs/connection-pooling.md:80]:
```suggestion
- When using a Digital Ocean connection pool, you'll have multiple ports available. Typically the direct connection (without connection pooling) is on port `25060` and the connection through pgbouncer is served through port `25061`. Make sure you connect to your connection pool on port `25061`
```

File: docs/docs/connection-pooling.md [hunk 7]
```diff
@@ -65,6 +65,22 @@ Heroku does not officially support MySQL.
...
+- Adjust the connection_limit: Clusters provide 25 connections per 1 GB of RAM. 3 connections per cluster are reserved for maintenance, and all remaining connections can be allocated to connection pools.
```
Comment (jtoar) [inline docs/docs/connection-pooling.md:81]:
```suggestion
- Adjust the `connection_limit`. Clusters provide 25 connections per 1 GB of RAM. Three connections per cluster are reserved for maintenance, and all remaining connections can be allocated to connection pools
```

File: docs/docs/connection-pooling.md [hunk 8]
```diff
@@ -65,6 +65,22 @@ Heroku does not officially support MySQL.
...
+- Both `pgbouncer=true` and `pool_timeout=30` are required in order to deploy succesfully through your connection pool
```
Comment (jtoar) [inline docs/docs/connection-pooling.md:82]:
```suggestion
- Both `pgbouncer=true` and `pool_timeout=30` are required to deploy successfully through your connection pool
```

**Review by jtoar — COMMENTED** [review]
Hey @rowin1125, made a few edits for readability. I think in summary the changes I'm suggesting are:
- reduce the number of "of" prepositions
- use `DATABASE_URL` instead of database url (cause it's an env var)
- "in order to" -> just "to"
- added some commas

Let me know what you think, otherwise this looks good!

**Comment by rowin1125** [comment]
@jtoar looks good to me, thanks 🙏

**Pull request merged**